File: Practice/src/coding_basic_training/FindByChangingString.js
function solution(myString, pat) {
    return myString.includes(
        pat.replaceAll('A', 'a').replaceAll('B', 'A').replaceAll('a', 'B')
    )
        ? 1
        : 0;
    // return [...myString]
    //     .map((v) => (v === 'A' ? 'B' : 'A'))
    //     .join('')
    //     .includes(pat)
    //     ? 1
    //     : 0;
}

File: Practice/src/coding_basic_training/FindByChangingString.java
package coding_basic_training;

public class FindByChangingString {
//	문자열 바꿔서 찾기
	public int solution(String myString, String pat) {
		pat = pat.replaceAll("A", "a");
		pat = pat.replaceAll("B", "A");
		pat = pat.replaceAll("a", "B");
		return myString.contains(pat) ? 1 : 0;
	}
}